django/thunderstore/api/cyberstorm/tests/test_packages.py

```diff
@@ -0,0 +1,562 @@
+@pytest.mark.django_db
+def test_packages_are_filtered_by_required_categories(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    cat1 = PackageCategory.objects.create(
+        name="c1", slug="c1", community=site.community
+    )
+    cat2 = PackageCategory.objects.create(
+        name="c2", slug="c2", community=site.community
+    )
+    PackageListingFactory(community_=site.community)
+    pl2 = PackageListingFactory(community_=site.community, categories=[cat1])
+    pl3 = PackageListingFactory(community_=site.community, categories=[cat2])
+
+    data = __query_api(api_client, f"included_categories={cat1.slug}")
+
+    __assert_packages_by_listings(data, pl2)
+
+    data = __query_api(
+        api_client,
+        f"included_categories={cat1.slug}&included_categories={cat2.slug}",
+    )
+
+    __assert_packages_by_listings(data, [pl3, pl2])
+
+
+@pytest.mark.django_db
+def test_packages_are_filtered_by_excluded_categories(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    cat1 = PackageCategory.objects.create(
+        name="c1", slug="c1", community=site.community
+    )
+    cat2 = PackageCategory.objects.create(
+        name="c2", slug="c2", community=site.community
+    )
+    pl1 = PackageListingFactory(community_=site.community)
+    PackageListingFactory(community_=site.community, categories=[cat1])
+    pl3 = PackageListingFactory(community_=site.community, categories=[cat2])
+
+    data = __query_api(api_client, f"excluded_categories={cat1.slug}")
+
+    __assert_packages_by_listings(data, [pl3, pl1])
+
+    data = __query_api(
+        api_client,
+        f"excluded_categories={cat1.slug}&excluded_categories={cat2.slug}",
+    )
+
+    __assert_packages_by_listings(data, pl1)
+
+
+@pytest.mark.django_db
+def test_packages_are_filtered_by_sections(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    required = PackageCategory.objects.create(
+        name="r", slug="r", community=site.community
+    )
+    excluded = PackageCategory.objects.create(
+        name="e", slug="e", community=site.community
+    )
+    irrelevant = PackageCategory.objects.create(
+        name="i", slug="i", community=site.community
+    )
+    section = PackageListingSection.objects.create(
+        name="Modpacks", slug="modpacks", community=site.community
+    )
+    section.require_categories.set([required])
+    section.exclude_categories.set([excluded])
+    expected = PackageListingFactory(community_=site.community, categories=[required])
+    PackageListingFactory(community_=site.community, categories=[required, excluded])
+    PackageListingFactory(community_=site.community, categories=[excluded])
+    PackageListingFactory(community_=site.community, categories=[irrelevant])
+
+    data = __query_api(api_client, f"section={section.slug}")
+
+    __assert_packages_by_listings(data, expected)
+
+
+@pytest.mark.django_db
+def test_section_does_not_exist(api_client: APIClient) -> None:
```

Comment:
What exactly is this testing? That when a non existing section is queried for, all listings are returned? Is this really the expected behaviour?

Reply:
fixd, to not return anything